tests/t520-operator.c

```diff
@@ -0,0 +1,268 @@
+    indxTet[i*2*PTet +  1] =  9 + offset;
+    indxTet[i*2*PTet +  2] = 16 + offset;
+    indxTet[i*2*PTet +  3] =  1 + offset;
+    indxTet[i*2*PTet +  4] =  8 + offset;
+    indxTet[i*2*PTet +  5] =  0 + offset;
+
+    indxTet[i*2*PTet +  6] = 14 + offset;
+    indxTet[i*2*PTet +  7] =  7 + offset;
+    indxTet[i*2*PTet +  8] =  0 + offset;
+    indxTet[i*2*PTet +  9] = 15 + offset;
+    indxTet[i*2*PTet + 10] =  8 + offset;
+    indxTet[i*2*PTet + 11] = 16 + offset;
+  }
+
+  // -- Restrictions
+  CeedElemRestrictionCreate(ceed, nelemTet, PTet, ndofs, dim, CEED_MEM_HOST,
+                            CEED_USE_POINTER, indxTet, &ErestrictxTet);
+  CeedElemRestrictionCreateIdentity(ceed, nelemTet, PTet, nelemTet*PTet, dim,
+                                    &ErestrictxiTet);
+
+  CeedElemRestrictionCreate(ceed, nelemTet, PTet, ndofs, 1, CEED_MEM_HOST,
+                            CEED_USE_POINTER, indxTet, &ErestrictuTet);
+  CeedElemRestrictionCreateIdentity(ceed, nelemTet, QTet, nqptsTet, 1,
+                                    &ErestrictuiTet);
+
+  // -- Bases
+  buildmats(qref, qweight, interp, grad);
+  CeedBasisCreateH1(ceed, CEED_TRIANGLE, dim, PTet, QTet, interp, grad, qref,
+                    qweight, &bxTet);
+
+  buildmats(qref, qweight, interp, grad);
+  CeedBasisCreateH1(ceed, CEED_TRIANGLE, 1, PTet, QTet, interp, grad, qref,
+                    qweight, &buTet);
+
+  // -- QFunctions
+  CeedQFunctionCreateInterior(ceed, 1, setup, __FILE__ ":setup", &qf_setupTet);
+  CeedQFunctionAddInput(qf_setupTet, "_weight", 1, CEED_EVAL_WEIGHT);
+  CeedQFunctionAddInput(qf_setupTet, "dx", dim, CEED_EVAL_GRAD);
+  CeedQFunctionAddOutput(qf_setupTet, "rho", 1, CEED_EVAL_NONE);
+
+  CeedQFunctionCreateInterior(ceed, 1, mass, __FILE__ ":mass", &qf_massTet);
+  CeedQFunctionAddInput(qf_massTet, "rho", 1, CEED_EVAL_NONE);
+  CeedQFunctionAddInput(qf_massTet, "u", 1, CEED_EVAL_INTERP);
+  CeedQFunctionAddOutput(qf_massTet, "v", 1, CEED_EVAL_INTERP);
+
+  // -- Operators
+  // ---- Setup Tet
+  CeedOperatorCreate(ceed, qf_setupTet, NULL, NULL, &op_setupTet);
+  CeedOperatorSetField(op_setupTet, "_weight", ErestrictxiTet, CEED_NOTRANSPOSE,
+                       bxTet, CEED_VECTOR_NONE);
+  CeedOperatorSetField(op_setupTet, "dx", ErestrictxTet, CEED_NOTRANSPOSE,
+                       bxTet, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_setupTet, "rho", ErestrictuiTet, CEED_NOTRANSPOSE,
+                       CEED_BASIS_COLLOCATED, qdataTet);
+  // ---- Mass Tet
+  CeedOperatorCreate(ceed, qf_massTet, NULL, NULL, &op_massTet);
+  CeedOperatorSetField(op_massTet, "rho", ErestrictuiTet, CEED_NOTRANSPOSE,
+                       CEED_BASIS_COLLOCATED, qdataTet);
+  CeedOperatorSetField(op_massTet, "u", ErestrictuTet, CEED_NOTRANSPOSE,
+                       buTet, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_massTet, "v", ErestrictuTet, CEED_NOTRANSPOSE,
+                       buTet, CEED_VECTOR_ACTIVE);
+
+  // Hex Elements
```

Comment:
The duplication feels tiresome, but I think it is largely due to the nature of the test I built.